docs/docs/data-structures/tree/schema-evolution/index.mdx

```diff
@@ -0,0 +1,219 @@
+- Removing an allowed type from a field
+- Renaming a node identifier
+
+
+### Checking Compatibility
+
+Use [`TreeView.compatibility`](../../../api/fluid-framework/treeview-interface#compatibility-property) to check if your schema changes are compatible with existing documents:
+
+```typescript
+const view = tree.viewWith(configuration);
+const { compatibility } = view;
+
+if (!compatibility.isEquivalent) {
+    if (compatibility.canUpgrade) {
+        view.upgradeSchema(); // Safe to upgrade
+    } else {
+        // Handle incompatible schema
+        throw new Error("Cannot open document with current schema");
+    }
+}
+```
+
+### Monitoring Remote Schema Changes
+
+In collaborative applications, other clients may upgrade the document's schema while you're working.
+Listen for the `schemaChanged` event to handle these remote upgrades:
+
+```typescript
+// Monitor for remote schema changes
+view.events.on("schemaChanged", () => {
+    if (!view.compatibility.canView) {
+        // Document was upgraded to an incompatible schema
+        // Handle gracefully - perhaps reload or notify user
+        throw new Error("Document was upgraded and is no longer compatible");
+    }
+
+    // Schema changed but is still compatible
+    // You may need to refresh UI or handle new data structures
+    console.log("Document schema was updated by another client");
+});
+```
+
+## Schema Upgrade Process
+
+When you make [backwards compatible](./index.mdx#understanding-compatibility) changes to your schema, you can upgrade existing documents to use the new schema using [`view.upgradeSchema()`](../../../api/fluid-framework/treeview-interface#upgradeschema-method):
+
+```typescript
+import { SchemaFactory, TreeViewConfiguration } from "@fluidframework/tree";
+
+const factory = new SchemaFactory("WhiteboardApp");
+
+// Updated schema with new optional field
+class Note extends factory.object("Note", {
+    id: factory.string,
+    x: factory.number,
+    y: factory.number,
+    color: factory.optional(factory.string), // New optional field
+}) {}
+
+// Create tree view with updated schema
+const config = new TreeViewConfiguration({ schema: Note });
+const view = tree.viewWith(config);
+
+// Check compatibility and upgrade if needed
+if (!view.compatibility.isEquivalent) {
+    if (view.compatibility.canUpgrade) {
+        view.upgradeSchema(); // Upgrade document to new schema
+        console.log("Document upgraded successfully");
+    } else {
+        throw new Error("Schema is incompatible and cannot be upgraded");
+    }
+}
+
+// Now you can use the upgraded document
+view.root.color = "#FF0000";
+```
+
+For detailed information about which types of changes are safe versus breaking, see [Types of Schema Changes](./types-of-changes).
+
+### Staged Rollouts
+
+When deploying schema changes in production, use staged rollouts to ensure compatibility between clients running different application versions.
+
+**Why Staged Rollouts Are Necessary**
+
```

Comment:
Let's emphasize here that staged rollouts allow you to handle changes that are backwards compatible but not forwards compatible, to connect it with the sections above.